src/Sylius/Bundle/UserBundle/Controller/UserController.php

```diff
@@ -105,6 +105,97 @@ public function resetPasswordAction(Request $request, $token)
         );
     }
 
+    /**
+     * @param Request $request
+     * @param string $email
+     * @param string $token
+     *
+     * @return Response
+     */
+    public function verifyAction(Request $request, $email, $token)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+        $redirectRoute = $request->attributes->get('_sylius[redirect]', 'sylius_shop_account_dashboard', true);
+        $response = $this->redirectToRoute($redirectRoute);
+
+        /** @var UserInterface $user */
+        $user = $this->repository->findOneByToken($token);
+        if (null === $user) {
+            if (!$configuration->isHtmlRequest()) {
+                return $this->viewHandler->handle($configuration, View::create($configuration, 400));
+            }
+
+            $this->addFlash('error', 'sylius.user.verification.error');
+
+            return $this->redirectToRoute('sylius_shop_homepage');
+        }
+        if (null !== $customer = $this->container->get('sylius.context.customer')->getCustomer()) {
+            if ($user !== $customer->getUser()) {
+                $securityToken = $this->container->get('security.token_storage')->getToken();
+                $this->container->get('security.logout.handler.session')->logout($request, $response, $securityToken);
+                $response->headers->remove('APP_REMEMBER_ME');
```

Comment:
No need to logout, when you login as different user the session will be overridden.

Btw. We should have scenario when I'm logged in as "ned@stark.com" and I verify account of "john@snow.com"